SchedulePlatform.Api/Controllers/AppointmentController.cs

```diff
@@ -1,81 +1,170 @@
+            _mapper = mapper;
+        }
+
+        [HttpGet("allApp/{nutritionistId}")]
+
+        public ActionResult<IEnumerable<AppointmentResponseModel>> GetAppoinmentByNutritionist(Guid nutritionistId)
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Success(_service.GetAppointmentByNutritionist(nutritionistId)));
+            }
+            catch (Exception ex)
+            {
+                if (_service.GetAll().ToList().FirstOrDefault(n => n.NutritionistId == nutritionistId) == null)
+                {
+                    return NotFound(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+                }
+                return BadRequest(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+        [HttpGet("all/{customerId}")]
+
+        public ActionResult<IEnumerable<AppointmentResponseModel>> GetAppointmentByCustomer(Guid customerId)
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Success(_service.GetAppointmentByCustomer(customerId)));
+            }
+            catch (Exception ex)
+            {
+                if (_service.GetAll().ToList().FirstOrDefault(c => c.CustomerId == customerId) == null)
+                {
+                    return NotFound(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+                }
+                return BadRequest(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+        [HttpGet]
+
+        public ActionResult<IEnumerable<AppointmentResponseModel>> GetAppointments()
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Success(_service.GetAll()));
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+        [HttpPost]
+
+        public IActionResult AddAppointment(AppointmentRequestModel requestappointment)
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<AppointmentResponseModel>.Success(_service.Add(requestappointment)));
+            }
+            catch (Exception ex)
+            {
+
+                return BadRequest(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+        [HttpGet("{id}")]
+
+        public IActionResult GetById(Guid id)
+        {
+            if (_service.GetById(id)==null)
+            {
+                return NotFound();
+            }
+            try
+            {
+                return Ok(ApiGenericsResult<AppointmentResponseModel>.Success(_service.GetById(id)));
```

Comment:
This logic looks a bit too complicated. Doing the exact same _service.GetById call before the try-catch block and inside it twice doesn't seem to efficient.